panel/widgets/tables.py

```diff
@@ -1629,9 +1625,10 @@ def _update_selection(self, indices: List[int] | SelectionEvent):
                 continue
             if selected:
                 ilocs.append(iloc)
-            else:
+            elif iloc in ilocs:
                 ilocs.remove(iloc)
         self.selection = list(dict.fromkeys(ilocs))
+        self.param.trigger('selection')
```

Comment:
I'm not entirely sure why this is needed to trigger the update.

Reply:
It shouldn't be unless the selection hasn't actually changed or something modified it in-place meaning that it can't detect the change.

Reply:
I was adding and removing from the `self.selection` list directly and therefore is the same on both sides of the equal sign on the line above.

I now copy the list of `self.selection` before adding/removing it so it registers a change.

Reply:
👍